### PR TITLE
Add with text color

### PR DESCRIPTION
**Pull request by alorma**
Based on #170 

Add `assertTextColor` and `assertNotTextColor`, as requested in #79 by @rocboronat

### CODE REVIEW

**Comment by Sloy** [comment]
I think your branch is 💩

**Comment by rocboronat** [comment]
Blocking this PR until #170 is merged, because it includes #170.

File: library/src/main/java/com/schibsted/spain/barista/assertion/BaristaVisibilityAssertions.kt [hunk 1]
```diff
@@ -70,4 +74,28 @@ object BaristaVisibilityAssertions{
...
+        onView(withText(containsString(text))).check(matches(not(isDisplayed())))
+    }
+
+    @JvmStatic
+    fun assertTextColor(@IdRes resId: Int, @ColorRes colorRes: Int) {
+        onView(withId(resId)).check(matches(TextColorMatcher(colorRes)))
+    }
+
+    @JvmStatic
+    fun assertNotTextColor(@IdRes resId: Int, @ColorRes colorRes: Int) {
```
Comment (Sloy) [inline library/src/main/java/com/schibsted/spain/barista/assertion/BaristaVisibilityAssertions.kt]: Something deep inside me tells me that `assertNotTextColor` is not very english-friendly. But I'm not sure how it should be 🤔

Comment (sergiandreplace) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaVisibilityAssertions.kt]: assertTextColorIs/assertTextColorIsNot or assertTextColorEquals/assertTextColorNotEqual?

Comment (Sloy) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaVisibilityAssertions.kt]: Like it!!

File: README.md [hunk 2]
```diff
@@ -169,11 +169,23 @@ assertHint(R.id.edittext, "Hint");
...
+
 // ...or not?
 assertNotContains("text");
 assertNotContains(R.id.textview, "text");
 
+// Check text is given color
+assertTextColorIs(R.id.textViewRed, R.color.textRed); // color
+assertTextColorIs(R.id.textViewBlue, R.color.textBlue); // color
+assertTextColorIs(R.id.textViewStateRed, R.color.text_state_red); // color state list
+assertTextColorIs(R.id.textViewStateBlue, R.color.text_state_blue); // color state list
```
Comment (Sloy) [inline README.md]: Are these duplicated? If the signature is the same, I think one single sample is enough.

Comment (alorma) [reply README.md]: Uooops

File: library/src/main/java/com/schibsted/spain/barista/internal/matcher/TextColorMatcher.kt [hunk 3]
```diff
@@ -15,7 +14,14 @@ class TextColorMatcher(@ColorRes private val color: Int) : BoundedMatcher<View,
     override fun matchesSafely(textView: TextView): Boolean {
         val resources = textView.context.resources
         colorName = resources.getResourceEntryName(color)
-        return textView.currentTextColor == ContextCompat.getColor(textView.context, color)
+        val colorMatches = textView.currentTextColor == ContextCompat.getColor(textView.context, color)
+        return if (colorMatches) {
+            true
+        } else {
+            val colorStateList = ContextCompat.getColorStateList(textView.context, color)
+            val colorForState = colorStateList.getColorForState(intArrayOf(-android.R.attr.enabled), colorStateList.defaultColor)
```
Comment (Sloy) [inline library/src/main/java/com/schibsted/spain/barista/internal/matcher/TextColorMatcher.kt]: Emmm what if the stat is not enabled? Or any other state? I guess this only works if the current color from the list is enabled or the default.

**Comment by Sloy** [comment]
I just pushed some changes:
- Simplified test scenarios to make it easier to understand the behavior. (Or so I think, sorry for the dictatorship here 😇 ). 
- Reproduced an issue where only enabled/disabled color states worked.
- Fixed the issue.
- Updated readme

Please @alorma take a quick look whenever you can to see if you agree with the solution. I think the feature is ready now.

**Comment by alorma** [comment]
Looks good for me @Sloy 

@rocboronat as you requested the feature, what do you think?

**Comment by rocboronat** [comment]
@alorma I think it's great! Everyone worked on this PR, it's the most collaborative one :·) Let's move!

**Comment by rocboronat** [comment]
Fuck. The readme was broken. The most collaborative PR and noone seen that the readme was showing the "yes" methods in te "not" section. Going to open a new PR 😂

**Comment by rocboronat** [comment]
By the way @alorma , try to give great names to the PR's to avoid commit messages like "Add with text color". Or, at less, give great commit messages when merging the PR's. Let's save the git history poetry!

**Pull request merged**